rts/src/main/java/eta/runtime/RuntimeOptions.java

```diff
@@ -0,0 +1,74 @@
+package eta.runtime;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class RuntimeOptions {
+    private Properties p;
+
+    public RuntimeOptions(String path) {
+        load_properties(path);
+    }
+
+    /*
+     * Loads properties from resource which
+     * may be override by system properties
+     */
+    private void load_properties(String path) {
+        Properties lp = new Properties();
+        // Load properties from resource
+        try (InputStream in = getClass().getClassLoader().getResourceAsStream(path)) {
+            if (in != null) lp.load(in);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        Properties sp = System.getProperties();
+        // Set loaded properties to be override by System properties
+        for (String key : lp.stringPropertyNames()) {
```

Comment:
`lp.stringPropertyNames()` will be empty if there's no properties file resource, so this will not override with System properties which is the expected behavior.

Reply:
Actually this is a comment I forgot to change here. What the code actually do is add properties loaded from resource to system poperties without overriding existing ones. Then use `sp` as the actual properties we read from.

Reply:
My bad, you're right! Thanks for the clarification.